dvc/fs/gdrive.py

```diff
@@ -208,15 +209,25 @@ def _drive(self):
         from pydrive2.auth import GoogleAuth
         from pydrive2.drive import GoogleDrive
 
-        if os.getenv(GDriveFileSystem.GDRIVE_CREDENTIALS_DATA):
-            with open(self._gdrive_user_credentials_path, "w") as cred_file:
+        temporary_save_path = self._gdrive_user_credentials_path
+        is_credentials_temp = os.getenv(
+            GDriveFileSystem.GDRIVE_CREDENTIALS_DATA
+        )
+        if self._use_service_account:
+            temporary_save_path = os.path.join(
+                tempfile.gettempdir(), "google-creds.json"
```

Comment:
The additional logic is needed because the saved file (cached authorization info) is different than the actual service json. The one that we have secret on the CI is the exported raw credentials json file. But the one pydrive's newer versions expect from this is the cached version. So we have to designate the temporary file as the raw credentials instead of the serialized ones.

Reply:
Oops, didn't realise that this is litterally `/tmp/google-creds.json`. Indeed, that's seems a bit odd.

Reply:
@isidentical Can we keep it in `.dvc/tmp` ?

Reply:
I didn't want to depend on `self.repo` in this logic (which is how we get `.dvc/tmp`, `self.repo.tmp_dir`) since we intend to remove `repo` access. Do you know a better way to get this?

Reply:
As I mentioned above - I see where it comes from, but I feel we don't need these changes. We can keep everything the way it was before all these changes (cache on CI saved credentials instead of the raw JSON file). Otherwise, if we want to support an env var we should introduce another one, since it has a different semantics.

Reply:
@isidentical `tmp_dir` will need to be passed as an option to the fs, it is fine. We might also switch to using proper global directories for it (e.g. using appdirs) in the future.

Reply:
@isidentical Wdyt about @shcheklein 's suggestion?

Reply:
>  cache on CI saved credentials instead of the raw JSON file

Can do it. This would require us to switch the CI variable to a serialized version of this raw file @efiop.

Reply:
> to a serialized version of this raw file

not raw but the one that GDrive saves into `.dvc/tmp/....`? right? just to be sure we are on the same page...

> Otherwise, if we want to support an env var we should introduce another one, since it has a different semantics.

I still like the idea behind this PR (to make it work with the raw file), @isidentical I think it's fine to create a ticket? It just feels that we need to do it bit differently (var name, path?, change docs, etc)

Reply:
> not raw but the one that GDrive saves into .dvc/tmp/....? right? just to be sure we are on the same page...

I guess we have to find proper names for these 2 different files since it might become very confusing (serialized file / raw file maybe? or cached file / raw file). Yes, that's what I meant by "serialized version of this raw file". 

> I still like the idea behind this PR (to make it work with the raw file), @isidentical I think it's fine to create a ticket? It just feels that we need to do it bit differently (var name, path?, change docs, etc)

My aim was make the the CI work, so I didn't see this as a feature though can create a ticket.